gapidapk/deviceinfo.go

```diff
@@ -99,6 +103,18 @@ func fetchDeviceInfo(ctx context.Context, d adb.Device) error {
 		return nil
 	}
 
+	if d.Instance().GetConfiguration().GetOS().GetAPIVersion() >= 29 {
+		startSignal, startFunc := task.NewSignal()
+		startFunc = task.Once(startFunc)
+		crash.Go(func() {
+			launchPerfettoProducerFromApk(ctx, d, startFunc)
+
+			// Make sure we fire the start signal.
```

Comment:
Please update the comment to "Ensure start signal is fired on failure/immediate return." I was confused at first why we signal the start at the end of the goroutine, as I thought it was the only place it is fired.